### ISSUE TITLE
Populate release as part of the `Step 1: Prep Release` workflow?

### ISSUE THREAD

**Issue by jtpio** [jupyter-server/jupyter_releaser#552]
### Problem

Running the `Step 1: Prep Release` is a good way to check the version is correctly bumped, and the changelog correctly created.

However this workflow does not populate the draft GitHub release with the built artifacts (wheel, sdist, npm tarballs), as this is done as part of the `populate-release` step in the `Step 2: Publish Release` workflow.

Maintainers might sometimes want to make sure the built assets are correct, and perform some manual checks.

### Proposed Solution

This likely does not require any change to the releaser core functionality. Maybe just moving this step:

https://github.com/jupyter-server/jupyter_releaser/blob/8568b9b5017b1a76e9c7af21315ecb04d933886b/example-workflows/publish-release.yml#L25-L32

To this workflow: https://github.com/jupyter-server/jupyter_releaser/blob/main/example-workflows/prep-release.yml

I guess one could try that on a test repo first to see if the whole flow would still work fine.

### Additional context

cc @martinRenou who I think was mentioning this after using the `python_packages` configuration of the releaser for https://github.com/voila-dashboards/voici

**Comment by blink1073** [comment]
Yeah, I don't see a way around that, since the assets are tied to the version bump commit.

**Comment by andrii-i** [comment]
Seeing built artifacts as a part of the prep release would help to diagnose problems with built artifacts like here #592 before they would fail at publish release preventing some assets from uploading.

**Comment by jtpio** [comment]
Maybe we could look into decoupling the assets upload from the commit to allow for this.

**Comment by krassowski** [comment]
For what it's wort, jupyter-scheduler artifacts are uploaded in `check-release` action, e.g. for the latest commit: https://github.com/jupyter-server/jupyter-scheduler/actions/runs/11809051201/job/32898683727. You need to click on "Summary" to see them: https://github.com/jupyter-server/jupyter-scheduler/actions/runs/11809051201

You can then go to `Build` action https://github.com/jupyter-server/jupyter-scheduler/actions/runs/11809051198/job/32898683728 again press "Summary" and see artifacts generated by hatch. These indeed differ in size.
